### ISSUE TITLE
Translations of texts in CDS views (DDLS) do not make it into the git repo

### ISSUE THREAD

**Issue by DerGuteWolf** [abapGit/abapGit#5297]
Seems to be missing currently in zcl_abapgit_object_ddls

**Comment by DerGuteWolf** [comment]
@mbtools  Could I temporarily set translation_languages (if not already set) with the help of ii_xml->i18n_params( ) and call serialize_lxe_texts() and afterwards empty translation_languages (if it was empty before) and similarily with deserialize?
Upon finish of #4470 we could simply remove the translation_languages calls afterwards.

**Comment by mbtools** [comment]
sounds like a hack ;-). not sure if there are side effects for example in parallel runs.

#4470 are follow-ups but it's already working in general. why not try the LXE logic straight up (maintain the LXE language in settings of course):

```
* serialize
    serialize_lxe_texts( io_xml ).

* deserialize
    deserialize_lxe_texts( io_xml ).
```

**Comment by DerGuteWolf** [comment]
@mbtools serialize works, but deserialize seems to conflict with the delete and recreate strategy used for DDLS, at the end of the deserialize method in zcl_abapgit_object_ddls the cds view does not exist and thus the texts can't be created.

**Comment by DerGuteWolf** [comment]
As a side note: In our EHP8 system the result from SYSTEM_INSTALLED_LANGUAGES and what is shown in SMLT differs (the FM returns approx double the number), in the S/4HANA system this matches.

For testing I used * for "Serialize Translations" in the EHP8 system which lead of cource to a bit more language entries than expected....

What should I use in "Serialize Translations", the SAP two letter codes, ie eg 3f (which is what you meant in https://github.com/abapGit/abapGit/issues/4470#issuecomment-770305929 , or?).

**Comment by mbtools** [comment]
in general we avoid object type specific code outside of the `zc_abapgit_object_xxxx` classes. maybe you can use the post processing exit until we find something better.

i added a comment to #4470. also sent something via slack.

**Comment by DerGuteWolf** [comment]
@mbtools yes, of course, this was just meant as a quick proof of concept that it will work. What do you think of a interface with the deserialize_lxe_texts like call, then we could stay with a loop at this position and test for implementation of the interface and then call the interface method?

**Comment by DerGuteWolf** [comment]
@mbtools I want to continue work on this. What do you think instead of `IF <ls_obj>-obj IS INSTANCE OF zcl_abapgit_object_ddls.` using a marker interface like `IF <ls_obj>-obj IS INSTANCE OF zif_abapgit_lxe_texts_postproc.`  and having a method `deserialize_lxe_texts` in the interface and call this method then?

**Comment by mbtools** [comment]
Sorry, I'm a bit short on time currently. Feel free to do a PR. Then we can better see the impact.

**Comment by DerGuteWolf** [comment]
@mbtools  Sorry, never got this really working and we have the problem when moving the cds views via transports, too. We have moved most of thei18n texts outside cds views now. This is why I haven't persued the problem here anymore

**Comment by mbtools** [comment]
I did some LXE improvements in #7232. Not sure if this helps in the cases you mentioned. Might be worth a test. It would be good to know what exactly works and what's missing still.

**Comment by mbtools** [comment]
https://github.com/abapGit/abapGit/pull/7232 moved the LXE translation handling after the object activation. Closing.
